JavaScript/TypeScript.sublime-syntax

```diff
@@ -887,7 +887,7 @@ contexts:
       scope: storage.modifier.abstract.js
 
     - match: import{{identifier_break}}
-      scope: keyword.operator.type.js
+      scope: keyword.operator.type.import.js
```

Comment:
A import a `keyword.operator`? Isn't `import` not just a `keyword.control`? I wouldn't scope it different from normal/other import keywords.

Reply:
This is a syntax for using imported types in a non-module context where you can't use `import` or `import type` declarations:

```ts
type Foo = import("./someModule").Foo;
```

I feel weird about the scoping. It's not really an operator (it's not operating on the `"./someModule"` type), but it's not really a control keyword either:

1. I'm actually a bit confused about `import` being a control keyword in the first place (although that's what the JS syntax currently does). I think that makes sense in some languages, where importing is essentially copy-pasting another file, or a higher-level language feature that loads and executes the other file in-line. In fairness, JS is such a language, but I wonder if it really generalizes?
2. TS has `import type from 'path'`, which definitely isn't a control keyword; the entire point of it is that it has no runtime effect, whereas even `import { type T } from 'path'` causes the module to be loaded. But it's basically the same syntax, so it probably should use the same scope even if it's not a perfect fit.
3. The `import("./someModule")` syntax definitely isn't a control keyword, and it also isn't just a variation on syntax that could be a control keyword. But it's not an operator either, and it probably should be at least similar to other import stuff.

I'm not sure how other syntaxes treat import keywords. I'm pretty sure that JS's `keyword.control.import-export` is not standard. If something like `keyword.import` were standard, it would make sense for all of the above.

Reply:
Loading something or just declaring an import, I second your concerns about `control` scope. `includes` and `imports` are more like general directives, which may be executed by a preprocessor, compiler or at runtime - depending on language.

Compiled languages like C or Erlang have various keyword being used in that non-control-flow context, which are "interpreted/evaluated" ahead of runtime (compile or import time). We should at least have a 2nd-level scope, which all of them could be grouped under. IIRC, there is an open RFC about that topic though. 

That said, I have a feeling we shouldn't push direction too much into semantic highlighting of keywords as we often can't reliably detect and scope them.